packages/theming/src/utils/componentStyles.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { DataAttributes, DefaultTheme, ThemeProps } from 'styled-components';
+
+/**
+ * CSS for component customizations based on `theme.components[componentId]`.
+ *
+ * @param {Object} props.theme Provides `components` object use to resolve the given component ID
+ * @param {String} [props.componentId] Specifies the lookup id for * `theme.components` styles.
+ *  The ID will be inferred from the `'data-garden-id'` attribute if not provided.
+ *
+ * @returns component CSS styles
+ */
+export const componentStyles = (props: ThemeProps<DefaultTheme> & { componentId?: string }) => {
+  let retVal: string | undefined;
+  const components = props.theme?.components;
```

Comment:
`ThemeProps` (or my suggestion ☝️) defines `theme` as [required](https://github.com/zendeskgarden/react-components/blob/main/utils/build/styled.d.ts). If this is what we want going forward:  
```suggestion
  const components = props.theme.components;
```